force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,41 @@
+trigger OpportunityTrigger on Opportunity (before update, before delete) {
+    
+    switch on Trigger.operationType {
+        when BEFORE_UPDATE {
+            
+            // Q7
+            Set<Id> accIds = new Set<Id>();
+            for (Opportunity opp : Trigger.new) {
+                accIds.add(opp.AccountId);                
+            }
+
+            Map<Id, Account> accMap = new Map<Id, Account> ([SELECT Id, (SELECT Id FROM Contacts WHERE Title = 'CEO') FROM Account WHERE Id IN :accIds]);
+            
+            for (Opportunity opp : Trigger.new) {
+                // Q5
+                if (opp.Amount <= 5000) {
+                    opp.addError('Opportunity amount must be greater than 5000');
+                }
+
+                // Q7
+                opp.Primary_Contact__c = accMap.get(opp.AccountId)?.Contacts[0].Id;
```

Comment:
What if there is no contacts that meets this criteria? You will get null pointer.

Reply:
It's hard to see (IMO) but I used the safe navigation operator after calling get on the map. In my anon apex testing if that were the case, it did return null instead of null pointer as designed. That OK?

Reply:
The safe navigator operator is only handling getting the account from the map, not in accessing the contacts records. It checks that there is an Account record in the map, not that then that account has contacts. So if there is no contact at index 0 (first contact), it would error.

Reply:
Yup, totally got it ok. Fixed with an if/then.